assets/shared/components/modal/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon, close as closeIcon } from '@wordpress/icons';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Modal component.
+ *
+ * @param {Object}   props
+ * @param {Function} props.onClose  Callback to run when trying to close the modal.
+ * @param {string}   props.title    The title for the modal. Empty by default.
+ * @param {Object}   props.children The content of the modal.
+ * @return {JSX.Element} The modal component or null (if not open).
+ */
+const Modal = ( { onClose, title = '', children } ) => {
+	useRunOnEscape( onClose );
+
+	return (
+		<div className={ 'sensei-modal' }>
+			<button
+				className="sensei-modal sensei-modal__overlay"
+				aria-label="Close"
+				onClick={ onClose }
+			/>
+			<div className="sensei-modal__wrapper">
+				<div className="sensei-modal__header">
+					<div className="sensei-modal__title">{ title }</div>
+					<button
+						className="sensei-modal sensei-modal__close-button"
+						onClick={ onClose }
+					>
+						<Icon icon={ closeIcon } />
+					</button>
+				</div>
+				<div className="sensei-modal__content">{ children }</div>
+			</div>
+		</div>
+	);
+};
+
+const useRunOnEscape = ( onClose ) => {
+	useEffect( () => {
+		const handleEsc = ( event ) => {
+			if ( event.keyCode === 27 ) {
+				onClose( event );
+			}
+		};
+		// Attach close event on Escape key.
+		// eslint-disable-next-line @wordpress/no-global-event-listener
+		window.addEventListener( 'keydown', handleEsc );
+
+		return () => {
+			// Detach from keydown on component unmounting.
+			// eslint-disable-next-line @wordpress/no-global-event-listener
+			window.removeEventListener( 'keydown', handleEsc );
```

Comment:
Same thing as above: I think if you add the event listener on `document.body`, you'll not need to disable eslint check.

Reply:
I was checking [the rule](https://github.com/WordPress/gutenberg/pull/26810), and I think the expected approach would be adding a ref in an element, and then call the `ownerDocument` from that.

I also just checked that Gutenberg is using a hook called [`useRefEffect`](https://github.com/WordPress/gutenberg/tree/trunk/packages/compose#userefeffect) for this type of events. See [an example here](https://github.com/WordPress/gutenberg/blob/fd0daedb128399bcb99d2e1829187ed0e70ebd2a/packages/block-editor/src/components/copy-handler/index.js#L159).

In this case, I think it would be something like this:

```js
const ref = useRefEffect( ( node ) => {
  const handleEsc = 
  ...
  node.ownerDocument.defaultView.addEventListener( 'keydown', handleEsc );
  ...
}, [ onClose ] );

return <div className="sensei-modal" ref={ ref }>...</div>
```

Reply:
Or maybe we don't even need add the listener to the window. I was checking that Gutenberg modal attaches it to [the overlay](https://github.com/WordPress/gutenberg/blob/v13.7.3/packages/components/src/modal/index.js#L132).

Reply:
Reimplemented the modal using the same approach than Gutenberg's modal: 0c029ec98

Reply:
asdfasdf